src/include/frag.h

```diff
@@ -211,6 +211,8 @@ class fragtree_t {
 	ls.front() == x &&
 	ls.size() == 1)
       return true;
+    // A node without leaves underneath it is definitely a leaf... 
+    if (ls.empty()) return true;
```

Comment:
How did you come across this? I don't know enough about behaviors here to know if this is a valid bugfix or not, but I'm concerned maybe these are allowed to be incomplete and not know the answers.